website/common/script/content/loginIncentives.js

```diff
@@ -145,13 +147,89 @@ module.exports = function getLoginIncentives (api) {
         user.items.food.Saddle += 1;
       },
     },
+    55: {
+      rewardKey: ['Pet_HatchingPotion_Purple'],
+      reward: [api.hatchingPotions.RoyalPurple],
+      assignReward: function assignReward (user) {
+        if (!user.items.hatchingPotions.RoyalPurple) user.items.hatchingPotions.RoyalPurple = 0;
```

Comment:
Yes, but let's delay the abstraction. I want to fix this in a more generic way. I'm postponing this until we redo some of the way we add content.